codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/protocol/MakeOperationGenerator.kt

```diff
@@ -138,43 +153,39 @@ open class MakeOperationGenerator(
     private fun buildOperationTypeRetry(writer: RustWriter, customizations: List<OperationCustomization>): String =
         customizations.mapNotNull { it.retryType() }.firstOrNull()?.let { writer.format(it) } ?: "()"
 
-    protected fun RustWriter.inRequestBuilderBaseFn(inputShape: StructureShape, f: RustWriter.() -> Unit) {
-        Attribute.Custom("allow(clippy::unnecessary_wraps)").render(this)
-        rustBlockTemplate(
-            "fn request_builder_base(input: &#{Input}) -> std::result::Result<#{HttpRequestBuilder}, #{OpBuildError}>",
-            *codegenScope,
-            "Input" to symbolProvider.toSymbol(inputShape)
-        ) {
-            f(this)
-        }
+    private fun needsContentLength(operationShape: OperationShape): Boolean {
+        return protocol.httpBindingResolver.requestBindings(operationShape)
+            .any { it.location == HttpLocation.DOCUMENT || it.location == HttpLocation.PAYLOAD }
     }
 
-    open fun generateRequestBuilderBaseFn(writer: RustWriter, operationShape: OperationShape) {
+    open fun createHttpRequest(writer: RustWriter, operationShape: OperationShape) {
         val httpBindingGenerator = RequestBindingGenerator(
             codegenContext,
             protocol,
             operationShape
         )
         val contentType = httpBindingResolver.requestContentType(operationShape)
-        val inputShape = operationShape.inputShape(codegenContext.model)
         httpBindingGenerator.renderUpdateHttpBuilder(writer)
-        writer.inRequestBuilderBaseFn(inputShape) {
-            Attribute.AllowUnusedMut.render(this)
-            writer.rust("let mut builder = update_http_builder(input, #T::new())?;", RuntimeType.HttpRequestBuilder)
-            val additionalHeaders = listOfNotNull(contentType?.let { "content-type" to it }) + protocol.additionalRequestHeaders(operationShape)
-            for (header in additionalHeaders) {
-                writer.rustTemplate(
-                    """
-                    builder = #{header_util}::set_request_header_if_absent(
-                        builder,
-                        #{http}::header::HeaderName::from_static(${header.first.dq()}),
-                        ${header.second.dq()}
-                    );
-                    """,
-                    *codegenScope
-                )
-            }
-            rust("Ok(builder)")
+
+        writer.rust("let mut builder = update_http_builder(&self, #T::new())?;", RuntimeType.HttpRequestBuilder)
+        if (includeDefaultPayloadHeaders && contentType != null) {
+            writer.rustTemplate(
+                "builder = #{header_util}::set_request_header_if_absent(builder, #{http}::header::CONTENT_TYPE, ${contentType.dq()});",
+                *codegenScope
+            )
+        }
+        for (header in protocol.additionalRequestHeaders(operationShape)) {
```

Comment:
are _these_ ones we want to always set? I guess so right? this is stuff like `operation` for JSON RPC?

Reply:
Yeah, it looks like it's only used by AWS JSON to add the `x-amz-target` header, which is needed to identify the operation to the service. We wouldn't want to remove these for this current use-case.